src/services/SearchService.ts

```diff
@@ -1,7 +1,281 @@
 import { Ask } from '../proto/ask';
+import facilityRepository, {
+  FacilityRepository
+} from '../repositories/FacilityRepository';
+import { Facility, Space } from '../proto/facility';
+import { DateTime, Interval } from 'luxon';
+import {
+  FacilityRuleRepository,
+  SpaceRuleRepository
+} from '../repositories/RuleRepository';
+import { DayOfWeekLOSRule, NoticeRequiredRule } from '../proto/lpms';
+import { FormattedDate, RulesItemKey } from './DBService';
+import { SpaceAvailabilityRepository } from '../repositories/SpaceAvailabilityRepository';
+import { SpaceStubRepository } from '../repositories/SpaceStubRepository';
+import ApiError from '../exceptions/ApiError';
+import { convertDaysToSeconds } from '../utils';
 
 export class SearchService {
```

Comment:
In future, maybe it would be best to refactor the `ask` into the `search` so that the `SearchService` can be initialized per faciltiy with the connectors etc retained in memory to reduce initialization / query time.

Reply:
The database connector is a singleton, I think that this will not affect performance